src/Rector/StaticCall/RouteActionCallableRector.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Laravel\Rector\StaticCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Identifier;
+use PHPStan\Reflection\Php\PhpMethodReflection;
+use PHPStan\Type\ObjectType;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\Reflection\ReflectionResolver;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Symplify\SmartFileSystem\SmartFileInfo;
+
+/**
+ * @see https://laravel.com/docs/8.x/upgrade#automatic-controller-namespace-prefixing
+ *
+ * @see \Rector\Laravel\Tests\Rector\StaticCall\RouteActionCallableRector\RouteActionCallableRectorTest
+ */
+final class RouteActionCallableRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    /**
+     * @var string
+     */
+    public const ROUTES = 'routes';
+
+    /**
+     * @var string
+     */
+    public const NAMESPACE = 'namespace';
+
+    private string $namespace = 'App\Http\Controllers';
+
+    /**
+     * @var array<string, string>
+     */
+    private array $routes = [];
+
+    public function __construct(
+        private ReflectionResolver $reflectionResolver,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Use PHP callable syntax instead of string syntax for controller route declarations.', [
+            new ConfiguredCodeSample(
+                <<<'CODE_SAMPLE'
+Route::get('/users', 'UserController@index');
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+Route::get('/users', [\App\Http\Controllers\UserController::class, 'index']);
+CODE_SAMPLE
+            ,
+                [
+                    self::NAMESPACE => 'App\Http\Controllers',
+                ]
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param Node\Expr\MethodCall|StaticCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isRouteAdditionMethod($node)) {
+            return null;
+        }
+
+        $position = $this->getActionPosition($node->name);
+
+        if (! isset($node->args[$position])) {
+            return null;
+        }
+
+        $arg = $node->args[$position];
+
+        $action = $this->valueResolver->getValue($arg->value);
+        if (! $this->isActionString($action)) {
+            return null;
+        }
+
+        [$controller, $method] = explode('@', $action);
```

Comment:
while technically never met, there should be a check for multiple `@`, eg: `Foo@bar@baz`, .so only applicable when string only have 1 `@` in the middle.

Reply:
Thanks for reminding, added a test fixture for that.